catalog/core/catalog-core-localstorageprovider/src/main/java/org/codice/ddf/catalog/content/impl/StorageProviderImpl.java

```diff
@@ -235,110 +240,128 @@ public DeleteStorageResponse delete(DeleteStorageRequest deleteRequest)
-                                    "Unable to commit changes for request: " + id);
-                        }
-                    }
-                }
-            } finally {
-                rollback(id);
-            }
+            commitUpdates(request, id, requestIdDir);
         } else if (deletionMap.containsKey(id)) {
-            List<Metacard> itemsToBeDeleted = deletionMap.get(id);
-            try {
-                for (Metacard metacard : itemsToBeDeleted) {
-                    LOGGER.debug("File to be deleted: {}", metacard.getId());
+            commitDeletes(request, id);
+        } else {
+            LOGGER.warn("Nothing to commit for request: {}", request.getId());
+        }
+    }
+
+    private void commitDeletes(StorageRequest request, String id) throws StorageException {
+        List<Metacard> itemsToBeDeleted = deletionMap.get(id);
+        try {
+            for (Metacard metacard : itemsToBeDeleted) {
+                LOGGER.debug("File to be deleted: {}", metacard.getId());
+
+                String metacardId = metacard.getId();
 
-                    String metacardId = metacard.getId();
+                String[] parts = getContentFilePathParts(metacardId);
 
-                    String[] parts = getContentFilePathParts(metacardId);
+                Path contentIdDir = Paths.get(baseContentDirectory.toAbsolutePath()
+                        .toString(), parts[0], parts[1], parts[2]);
 
-                    Path contentIdDir = Paths.get(baseContentDirectory.toAbsolutePath()
-                            .toString(), parts[0], parts[1], parts[2]);
+                if (!Files.exists(contentIdDir)) {
+                    throw new StorageException("File doesn't exist for id: " + metacard.getId());
+                }
 
-                    if (!Files.exists(contentIdDir)) {
-                        throw new StorageException(
-                                "File doesn't exist for id: " + metacard.getId());
+                try {
+                    List<Path> files = listPaths(contentIdDir);
+                    for (Path file : files) {
+                        Files.deleteIfExists(file);
+                    }
+                    Files.deleteIfExists(contentIdDir);
+                    Path part1 = contentIdDir.getParent();
+                    List<Path> part1Files = listPaths(part1);
+                    if (part1Files.size() == 0) {
+                        Files.deleteIfExists(part1);
                     }
+                    Path part0 = part1.getParent();
+                    List<Path> part0Files = listPaths(part0);
+                    if (part0Files.size() == 0) {
+                        Files.deleteIfExists(part0);
+                    }
+                } catch (IOException e) {
+                    throw new StorageException("Could not delete file: " + metacard.getId(), e);
+                }
+            }
+        } finally {
+            rollback(request);
```

Comment:
I understand that we need to rollback on failure, but why do we rollback on success?